xarray/core/coordinates.py

```diff
@@ -664,6 +664,8 @@ def copy(
         variables = {
             k: v._copy(deep=deep, memo=memo) for k, v in self.variables.items()
         }
+        # Currently this requires using the concrete type of `Coordinates`, rather than
+        # `Self`, ref #8216
```

Comment:
why doesn't type(self) work here?
```python
        return type(self)._construct_direct(
            coords=variables, indexes=dict(self.xindexes), dims=dict(self.sizes)
        )
```

Reply:
I went to answer this question, and then got it working :)

Reply:
OK actually the typing worked but the tests didn't; switching back. Added a TODO but won't chase it down atm...